memeolist/app/src/main/java/org/aerogear/android/app/memeolist/ui/MemeFormActivity.java

```diff
@@ -174,21 +164,23 @@ private void saveMeme(String imageUrl) {
                 .photourl(createMemeUrl(imageUrl))
                 .build();
 
-        apolloClient.mutate(mutation)
-                .enqueue(new ApolloCall.Callback<CreateMemeMutation.Data>() {
+        SyncClient
+                .getInstance()
+                .mutation(mutation)
+                .execute(CreateMemeMutation.Data.class)
+                .respondOn(new AppExecutors().mainThread())
+                .respondWith(new Responder<Response<CreateMemeMutation.Data>>() {
                     @Override
-                    public void onResponse(@Nonnull Response<CreateMemeMutation.Data> response) {
-                        new AppExecutors().mainThread().submit(() -> {
-                            progress.dismiss();
-                            displayMessage(R.string.meme_created);
-                            finish();
-                        });
+                    public void onResult(Response<CreateMemeMutation.Data> response) {
+                        progress.dismiss();
```

Comment:
We are missing error handler here.

Reply:
The error handler is automatically handled by the SDK. See: https://github.com/aerogear/aerogear-android-sdk/blob/master/sync/src/main/java/org/aerogear/mobile/sync/SyncClient.java#L171-L179